src/editor/validator.ts

```diff
@@ -381,23 +381,51 @@ export class Validator {
             context.lineStatement.rootNode instanceof Statement
         );
     }
+    /** 
+     * logic: checks if token is not null AND instanceof BinaryOperatorExpr AND atEmptyExpressionHole
+     */
+    isBinaryOperatorExprTknEmpty(providedContext?: Context):boolean{
+        const context = providedContext ? providedContext : this.module.focus.getContext();
+
+        if(context.token === null) return false;
+        if(!(context.token.rootNode instanceof BinaryOperatorExpr)) return false;
+        if(!(this.atEmptyExpressionHole)) return false;
+
+        return true
+    }
+
+    /**
+     * logic: checks if both tokens of BinaryOperatorExpr are empty
+     */
+
+    isAllBinaryOperatorExprTknsEmpty(providedContext? :Context):boolean{
+        const context = providedContext? providedContext : this.module.focus.getContext();
+        if(context.token.rootNode instanceof BinaryOperatorExpr){
+            let leftOperand = context.token.rootNode.getLeftOperand();
+            let rightOperand = context.token.rootNode.getRightOperand();
+            if(leftOperand instanceof TypedEmptyExpr && rightOperand instanceof TypedEmptyExpr){
+                return true;
+            }
+        }
+        
+        return false;
+    }
 
     /**
+     * 
      * logic: checks if at the end of a statement, and not text editable.
      * AND does not have a body.
      * AND prev item is not an expression that could be deleted by it self.
      */
     canDeletePrevStatement(providedContext?: Context): boolean {
         const context = providedContext ? providedContext : this.module.focus.getContext();
-
```

Comment:
please also make sure that you do not remove the spaces between these. please refer to typescript's style guide by google: https://google.github.io/styleguide/tsguide.html